### PR TITLE
Implement pan (unbounded)

### PR DESCRIPTION
**Pull request by axelboc**

### CODE REVIEW

File: src/h5web/visualizations/heatmap/Mesh.tsx [hunk 1]
```diff
@@ -13,23 +12,13 @@ function Mesh(props: Props): JSX.Element {
...
-    evt.stopPropagation();
-
-    // Fix react-three-fiber typings
-    const camera = evt.camera as OrthographicCamera;
-    const wheelEvt = evt as React.WheelEvent<HTMLDivElement>;
-
-    const factor = wheelEvt.deltaY > 0 ? ZOOM_FACTOR : 1 / ZOOM_FACTOR;
-    camera.zoom = Math.max(1, camera.zoom * factor);
-    camera.updateProjectionMatrix();
-  }
```
Comment (axelboc) [inline src/h5web/visualizations/heatmap/Mesh.tsx:29]: I move this handler inside the new custom hook that does both zoom and pan.

File: src/h5web/visualizations/heatmap/utils.ts [hunk 2]
```diff
@@ -23,3 +28,67 @@ export function computeTextureData(matrix: number[][]): Uint8Array | undefined {
...
+
+  const onPointerDown = useCallback(
+    (evt: PointerEvent) => {
+      evt.stopPropagation();
+
+      const { currentTarget, pointerId } = evt as React.PointerEvent;
+      currentTarget.setPointerCapture(pointerId);
+
+      const projectedPoint = camera.worldToLocal(evt.unprojectedPoint.clone());
+      setStartOffsetPosition(camera.position.clone().add(projectedPoint));
```
Comment (axelboc) [inline src/h5web/visualizations/heatmap/utils.ts:43]: When dragging starts, I save a **start offset position**, which is the sum of the positions of the camera and of the mouse pointer (projected from world to camera space). When the pointer moves during the drag, I compute the camera's new position by subtracting the pointer's projected position from the start offset position.

File: src/h5web/visualizations/heatmap/utils.ts [hunk 3]
```diff
@@ -23,3 +28,67 @@ export function computeTextureData(matrix: number[][]): Uint8Array | undefined {
...
+    },
+    [camera]
+  );
+
+  const onPointerUp = useCallback((evt: PointerEvent) => {
+    evt.stopPropagation();
+
+    const { currentTarget, pointerId } = evt as React.PointerEvent;
+    currentTarget.releasePointerCapture(pointerId);
```
Comment (axelboc) [inline src/h5web/visualizations/heatmap/utils.ts:52]: Pointer capturing allows pointer events to remain targeted to the initial target object (i.e. the mesh) throughout the drag, event when the pointer moves outside of the bounds of the mesh (e.g. when it moves onto the explorer sidebar).

File: src/h5web/visualizations/heatmap/utils.ts [hunk 4]
```diff
@@ -23,3 +28,67 @@ export function computeTextureData(matrix: number[][]): Uint8Array | undefined {
...
+  const onPointerMove = useCallback(
+    (evt: PointerEvent) => {
+      if (!startOffsetPosition.current) {
+        return;
+      }
+
+      evt.stopPropagation();
+
+      const projectedPoint = camera.worldToLocal(evt.unprojectedPoint.clone());
+      const { x: pointX, y: pointY } = projectedPoint;
```
Comment (loichuder) [inline src/h5web/visualizations/heatmap/utils.ts:66]: The two above lines cannot be put in one ?

Comment (axelboc) [reply src/h5web/visualizations/heatmap/utils.ts:66]: It can, but it then wraps onto three lines... 😂 Also, I thought the variable added clarity.

**Comment by axelboc** [comment]
> Amazingly concise 👍

Thanks 😊

**Pull request merged**